### PR TITLE
#1668 - invitation as expert bug

### LINKED ISSUE #1667: [Access request] Remove Accept / Refuse access request from creation

### LINKED ISSUE #1668: [Study access] Add user to study using its login as expert
isMeSecurityService.isMe(userName)  -> Cannot work properly as an expert

### PR DESCRIPTION
**Pull request by jcomedouteau**
- Add specific method for invitation to get user by username

Closes #1668 

How to test:
- Connect as an expert
- Go to an administrable study detail
- Edit it to add a member using its username (and not mail)
--> The user is well added

### CODE REVIEW

**Comment by youennmerel** [comment]
Hi @jcomedouteau,

Adding a user by his login is OK for me in dev.

When trying to add a non existing login or mail, the following error occurs : "An internal error occured while adding the user. Please try again later or contact an administrator. "

I think we should rather explicitly indicate that the chosen username / email does not exists in this case.

However, this is absolutely non blocking, and we can merge anyway if time is of the essence !

File: shanoir-ng-users/src/main/java/org/shanoir/ng/accessrequest/controller/AccessRequestApiController.java [hunk 1]
```diff
@@ -194,7 +194,7 @@ public 	ResponseEntity<AccessRequest> inviteUserToStudy(
 		Optional<User> user = this.userService.findByEmail(email);
 		
 		if (!user.isPresent()) {
-			user = this.userService.findByUsername(email);
+			user = this.userService.findByUsernameForInvitation(email);
```
Comment (youennmerel) [inline shanoir-ng-users/src/main/java/org/shanoir/ng/accessrequest/controller/AccessRequestApiController.java:197]: Maybe rename the "email" variable ?

**Comment by jcomedouteau** [comment]
Actually the login should not contain any '@' char, so I was thinking of just checking it..

**Comment by michaelkain** [comment]
Hi, one option could be to make it the other way round, check for a valid email and try otherwise, see
https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/EmailValidator.html Just an idea.

**Review by youennmerel — APPROVED** [review]
Hi @jcomedouteau,

Tested in qualif, it seems to work all right.

**Pull request merged**